bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ProjectEncodingPreferenceChangeListener.java

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ *
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License 2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Ingo Mohr - Issue #166 - Add Preference to Turn Off Warning-Check for Project Specific Encoding
+ *******************************************************************************/
+package org.eclipse.core.internal.resources;
+
+import java.util.*;
+import org.eclipse.core.internal.utils.Messages;
+import org.eclipse.core.resources.*;
+import org.eclipse.core.runtime.*;
+import org.eclipse.core.runtime.jobs.Job;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.IPreferenceChangeListener;
+import org.eclipse.core.runtime.preferences.IEclipsePreferences.PreferenceChangeEvent;
+import org.eclipse.core.runtime.preferences.IScopeContext;
+import org.osgi.service.component.annotations.*;
+
+@Component(service = {})
+public class ProjectEncodingPreferenceChangeListener implements IPreferenceChangeListener {
+
+	private static final String MARKER_TYPE = ValidateProjectEncoding.MARKER_TYPE;
+
+	@Reference
+	private IWorkspace workspace;
+
+	@Reference
+	private ILog log;
+
+	@Reference(target = IScopeContext.BUNDLE_SCOPE_FILTER)
+	IScopeContext bundleScope;
+
+	private IEclipsePreferences eclipsePreferences;
+
+	@Activate
+	public void register() {
+		eclipsePreferences = bundleScope.getNode(""); //$NON-NLS-1$
+		eclipsePreferences.addPreferenceChangeListener(this);
+	}
+
+	@Deactivate
+	public void unregister() {
+		eclipsePreferences.removePreferenceChangeListener(this);
+	}
+
+	@Override
+	public void preferenceChange(PreferenceChangeEvent event) {
+		if (ResourcesPlugin.PREF_MISSING_ENCODING_MARKER_SEVERITY.equals(event.getKey())) {
+			final int newSeverity = event.getNewValue() != null ? Integer.parseInt((String) event.getNewValue())
+					: PreferenceInitializer.PREF_MISSING_ENCODING_MARKER_SEVERITY_DEFAULT;
+			final int oldSeverity = event.getOldValue() != null ? Integer.parseInt((String) event.getOldValue())
+					: PreferenceInitializer.PREF_MISSING_ENCODING_MARKER_SEVERITY_DEFAULT;
+
+			workspace = ResourcesPlugin.getWorkspace();
+
+			if (isIgnore(newSeverity)) {
+				removeAllRelatedMarkers(workspace.getRoot());
+			} else if (newSeverity != oldSeverity) {
+				updateAllRelatedMarkers(workspace.getRoot(), newSeverity);
+			}
+		}
+	}
+
+	private void removeAllRelatedMarkers(IContainer workspaceRootOrProject) {
+		final Collection<IMarker> markers = getRelatedMarkers(workspaceRootOrProject);
+		if (markers.isEmpty()) {
+			return;
+		}
+		WorkspaceJob job = new WorkspaceJob(Messages.updateMissingProjectEncodingMarkers) {
```

Comment:
I suppose we could batch all the projects into a single job.  Though I'm not sure that's actually better...

Reply:
I might be missing something here, but: Aren't we already batching all the projects into a single job, here? The method is called w/ workspace.root for which all markers are resolved that are updated all w/in the same one job starting at line 77.

Reply:
Ah, maybe you mean the other job down below on line 121? That one is called for each of the projects. However, I can't tell which option is better in terms of performance or RAM-usage.

Reply:
Given that changing the preference isn't a common operation I think we shouldn't fuss over the performance...